cpp/src/arrow/array/data.h

```diff
@@ -180,25 +180,46 @@ struct ARROW_EXPORT ArrayData {
 
   std::shared_ptr<ArrayData> Copy() const { return std::make_shared<ArrayData>(*this); }
 
-  bool IsNull(int64_t i) const { return !IsValid(i); }
+  inline bool IsNull(int64_t i) const { return !IsValid(i); }
 
-  bool IsValid(int64_t i) const {
+  inline bool IsValid(int64_t i) const {
     if (buffers[0] != NULLPTR) {
       return bit_util::GetBit(buffers[0]->data(), i + offset);
     }
     const auto type = this->type->id();
     if (type == Type::SPARSE_UNION) {
       return !internal::IsNullSparseUnion(*this, i);
-    }
-    if (type == Type::DENSE_UNION) {
+    } else if (type == Type::DENSE_UNION) {
```

Comment:
nit: this line of change seems to be complained by clang-tidy.

Reply:
Not on my editor. Where did you see this?

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html

Reply:
In this specific function, the use of `else` is not creating unnecessary indentation levels.

Reply:
Sounds good. I was overwhelmed by this kind of warning in the internal toolchain from my employer. :)